.github/workflows/first_issue_reply.yml

```diff
@@ -18,7 +18,18 @@ jobs:
       - name: Checkout main repository
         uses: actions/checkout@v4
 
+      - name: Check for existing bot comments
+        id: check-comments
+        run: |
+          ISSUE_NUMBER=${{ github.event.issue.number }}
+          BOT_COMMENT_EXISTS=$(gh issue view "$ISSUE_NUMBER" --json comments --jq '.comments[].body' | grep -q "Hi @$USERNAME," && echo "true" || echo "false")
```

Comment:
My command line is giving me issues with parsing the single quote on the jq expression. Are you certain this works?

Reply:
I ran it in my wsl and also wrote a test file I think it is working fine. You can also check my repo/Create a repo and replace the first issue reply file with this and see if it works. My repo: https://github.com/ZheyuDeng/testing-github-action/issues/4

Below is my test file and result

![image](https://github.com/Submitty/Submitty/assets/122962727/c419a8c2-ef5e-4011-87f0-60da76b57ec5)
![image](https://github.com/Submitty/Submitty/assets/122962727/fd5984dc-3aa4-4383-a50a-046c06d1a6b1)